prog/vm/nexus.rb

```diff
@@ -167,7 +167,14 @@ def storage_secrets
 
   def allocation_dataset
     requires_bdev_ubi = frame["storage_volumes"].any? { |v| v["use_bdev_ubi"] }
-    spdk_where_clause = requires_bdev_ubi ? "AND id in (select vm_host_id from spdk_installation where version like '%ubi%' and allocation_weight > 0)" : ""
+    # Currently, github runners require btrfs if not using bdev_ubi. We will use
+    # ext4 for hosts with bdev_ubi installation. So, we make sure we don't
+    # schedule disks on a bdev_ubi enabled host if bdev_ubi is not used.
+    #
+    # YYY: revisit this check after migrating away from btrfs, or having some
+    # direct way to check for filesystem type.
+    version_qualifier = requires_bdev_ubi ? "like '%ubi%'" : "not like '%ubi%'"
```

Comment:
a bit puzzled about this, because status quo is the only two "version" available is "v23.09" and "LEGACY_SPDK_VERSION", how do `ubi` substrings get into the picture?

Reply:
Versions available are:
* `LEGACY_SPDK_VERSION`
* `v23.09` (built from stock SPDK, our default)
* `v23.09-ubi-0.2` (package created by bdev_ubi and uploaded to https://github.com/ubicloud/bdev_ubi/releases

See https://github.com/ubicloud/ubicloud/blob/69baeddb9e0b7e49c52f473209ccfb072f64fe30/prog/storage/setup_spdk.rb#L7-L9